app/domain/streams/handlers/multipart_handler.rb

```diff
@@ -12,11 +12,20 @@ def initialize(attr_list, content_id, locale, payload, routing_key)
   def process
     base_paths = attr_list.map { |hsh| hsh[:base_path] }
     deprecate_redundant_paths(base_paths)
-    update_editions(attr_list.map(&method(:find_old_edition)))
+    current_editions = update_editions(attr_list.map(&method(:find_old_edition)))
+    set_parent_relationships(current_editions)
   end
 
 private
 
+  def set_parent_relationships(current_editions)
+    parent = current_editions.first
+    children = current_editions.drop(1)
```

Comment:
Can we simplify the variable `current_editions` to `editions` for this function? - theres no need to distinguish between current and previous